common/lib/xmodule/xmodule/library_tools.py

```diff
@@ -96,6 +96,17 @@ def _filter_child(self, usage_key, capa_type):
         assert isinstance(descriptor, CapaDescriptor)
         return capa_type in descriptor.problem_types
 
+    def can_use_library_content(self, course_id):
+        """
+        Determines whether a modulestore holding a course_id supports libraries.
+        """
+        store = self.store
+        # Skip MixedModuleStore abstraction.
+        if hasattr(store, '_get_modulestore_for_courseid'):
+            # pylint: disable=protected-access
+            store = store._get_modulestore_for_courselike(course_id)
+        return hasattr(store, "copy_from_template")
```

Comment:
Generally speaking, this entire file (library_tools.py) makes me uncomfortable in the amount that it assumes that the `LibraryContentModule` should know about how the containing application works. In particular, `modulestore` isn't a concept/interface that appears in XBlock.

If a particular modulestore can't support libraries, perhaps it would be more appropriate to make those modulestores explicitly filter the library block out? Or explictly raise an error when asked to create that type of block?

Separately, rather than having `get_library` as part of the service interface, it seems like maybe the `LibraryContentModule` should just be able to call `runtime.get_block` and pass in the library key (as that's the interface that a block would use to load some other block normally).